datastore/src/main/scala/org/allenai/datastore/Datastore.scala

```diff
@@ -382,6 +384,31 @@ class Datastore(val name: String, val s3: AmazonS3Client) extends Logging {
   // Putting data into the datastore
   //
 
+  private def multipartUpload(path: Path, locator: Locator): Unit = {
+    val tm = new TransferManager(s3)
+    try {
+      val request = new PutObjectRequest(bucketName, locator.s3key, path.toFile)
+      request.setGeneralProgressListener(new ProgressListener {
+        private var lastLogMessage = System.currentTimeMillis()
+
+        override def progressChanged(progressEvent: ProgressEvent): Unit = {
+          val now = System.currentTimeMillis()
+          if (now - lastLogMessage >= 1000) {
+            logger.info(
+              s"Uploading $path to the $name datastore. " +
```

Comment:
Seems like you could log this at the start, and just log the progress in the listener. Not sure how long path part of this message is, though.